src/main/java/org/opensearch/performanceanalyzer/reader_writer_shared/EventLogFileHandler.java

```diff
@@ -113,11 +115,19 @@ public void renameFromTmp(long epoch) {
     public void renameFromTmpWithPrivilege(long epoch) {
         Path path = Paths.get(metricsLocation, String.valueOf(epoch));
         Path tmpPath = Paths.get(path.toString() + TMP_FILE_EXT);
-        // This is done only when no exception is thrown.
-        try {
-            Files.move(tmpPath, path, REPLACE_EXISTING, ATOMIC_MOVE);
-        } catch (IOException e) {
-            LOG.error("Error moving file {} to {}.", tmpPath.toString(), path.toString(), e);
+        File tempFile = new File(tmpPath.toString());
+        // Only if the tmp file is present, we want to create the writer file.
+        // If not, we will publish a metric.
+        if (tempFile.exists()) {
```

Comment:
We should check if the tmpfile exists and also has data ?

Reply:
We create the tmp file only if we data. tmp file creation is skipped if the queue is empty.